vendor/k8s.io/kubernetes/pkg/kubectl/cmd/create.go

```diff
@@ -150,7 +152,21 @@ func RunCreate(f *cmdutil.Factory, cmd *cobra.Command, out io.Writer, options *C
 		}
 
 		count++
-		shortOutput := cmdutil.GetFlagString(cmd, "output") == "name"
+		shortOutput := output == "name"
+		if len(output) > 0 && !shortOutput {
+			printer, generic, err := cmdutil.PrinterForCommand(cmd)
+			if err != nil {
+				return err
+			}
+			if !generic {
+				decodedObj, decoded := decodeUnstructuredObject(info.Object)
+				if !decoded {
+					return fmt.Errorf("Unsupported output format for the current object")
+				}
+				return f.PrintObject(cmd, registered.RESTMapper(), decodedObj, out)
```

Comment:
@fabianofranz Added support for printing unstructured objects with a wide printer. PTAL
Sample output:

```
$ oc create -f pkg/api/graph/test/bc-missing-output.yaml --dry-run -o wide
NAME          TYPE      FROM      LATEST
gitauthtest   Source    Git       1
```